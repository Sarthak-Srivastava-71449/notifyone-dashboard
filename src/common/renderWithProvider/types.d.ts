/*
    Auth User Types
*/

interface IUser {
  loading: boolean;
  error: string;
  success: boolean | number;
  name: string;
  tokens: {
    accessToken: string;
  };
  user_info: {
    email_id: string;
    contact_number: string;
    email_verified: boolean;
  };
  roles: IRoles;
}

type UserAction = {
  type?: string;
  payload?: IUser;
};

type UserActionDispatchType = (args: UserAction) => UserAction | any;

interface AxiosGoogleResponse {
  auth_token: string;
  auth_token_labs: string;
  token: string;
  username: string;
}

interface UserInfoResponse {
  roles: IRoles;
  user_info: {
    email_id: string;
    contact_number: string;
    email_verified: boolean;
  };
}
