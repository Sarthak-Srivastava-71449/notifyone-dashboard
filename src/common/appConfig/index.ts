const AppConfig: any = process.env.communication;

export default AppConfig;
