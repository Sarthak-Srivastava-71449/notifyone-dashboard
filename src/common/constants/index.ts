export const COMMUNICATION_TYPE = {
  All: 'All',
  SMS: 'SMS',
  Whatsapp: 'Whatsapp',
  Transactional_Push_notification: 'Transactional Push Notification',
  Email: 'Email',
};

export const MAX_INPUT_LIMIT = 30;

export const AUTH_ROLES = 'LARA_ADMIN';

export const EVENT_FETCH_LIMIT = 50;

// Navigation
export const COMS_TEMPLATE_NAVIGATE = '/communication/template';
