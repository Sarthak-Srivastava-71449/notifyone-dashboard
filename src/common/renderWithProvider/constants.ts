export const mockEmailEventDetails = {
  actions: 1,
  app_name: 'off',
  description: 'ccq_orders_time_reporting',
  event_id: 207,
  event_name: 'ccq_orders_time_reporting',
  event_text: 'p PFA',
  event_type: 'Email',
  id: 183,
  includes: [],
  name: 'ccq_orders_time_reporting',
  subject: 'CCQ Reporting',
  triggers_limit: -1,
  updated_by: 'rohit.taneja@1mg.com',
};

export const mockSmsEventDetails = {
  actions: 1,
  app_name: 'Doctors',
  event_name: 'user_cancelled_appointment',
  event_text:
    'Dear {{patient}},\nYour consultation with {{doctor_name}} at {{time}}, {{date}} has been cancelled.\nRefund has been initated to your source account. It will reflect within 5-7 working days. We are sorry to see you go.\nThanks. -Tata 1mg',
  event_type: 'SMS',
  id: 504,
  triggers_limit: -1,
};

export const mockPushNotificationEventDetails = {
  actions: 1,
  app_name: 'offer_service',
  event_name: 'last_days_before_expiry',
  event_text: '{{body}}',
  event_type: 'Transactional Push Notification',
  id: 83,
  image: null,
  target: 'DYNAMIC TARGET',
  title: '{{title}}',
  triggers_limit: -1,
  updated: 1629096532,
  updated_by: 'lara_service_user',
};

export const mockWhatsappEventDetails = {
  actions: 0,
  app_name: 'diagnostics',
  event_name: 'report_discripency_whatsapp',
  event_text: '32',
  event_type: 'Whatsapp',
  id: 180,
  triggers_limit: -1,
};
export const mockRavenRootNodeEvents = {
  result: [
    {
      node_details: {
        sub_questions: [
          {
            sub_questions: [
              {
                updated_by: 'v-animesh.singh@1mg.com',
                event: null,
                required_fields: ['free_text'],
                emails: null,
                screen_text: null,
                optional_fields: null,
                type: 'FORM',
                node_name: '1.161.159',
                id: '1.161.159',
                header: 'Feedback',
                node_action: 73,
                response:
                  '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.',
                name: 'What is 1mg?',
                ticket: {
                  status: 'OPEN',
                  priority: 'MEDIUM',
                  updated_by: 'initialdbsetup@1mg.com',
                  id: 3,
                },
              },
              {
                updated_by: 'ankita.gupta@1mg.com',
                event: null,
                required_fields: null,
                emails: null,
                screen_text:
                  'You can easily reach out to us via email: <br>\n\n~ Any query related to Medicine orders, please click on <b><a href=https://www.1mg.com/order/history =>Order History</a> </b> then on Need Help. <br>\n~ Any query related to Lab Tests, please click on <b><a href=https://www.1mg.com/labs?utm_source=1mg&utm_medium=jewel&utm_campaign=labsgrowth =>Lab Tests </a> </b> then on Need Help on the right side corner. <br>\n~ Any query related to Doctor Consultation, click here <b><a href=https://www.1mg.com/online-doctor-consultation =>Online Consultation</a> </b>',
                optional_fields: null,
                type: 'INFO',
                node_name: '1.161.168',
                id: '1.161.168',
                header: 'How to contact us?',
                node_action: 176,
                response:
                  '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id within next 24 hours.',
                name: 'How to contact 1mg?',
                ticket: {
                  status: 'OPEN',
                  priority: 'LOW',
                  updated_by: 'initialdbsetup@1mg.com',
                  id: 2,
                },
              },
            ],
          },
        ],
        node_name: '466',
        id: '466',
        node_action: null,
        name: 'Cancellation Initiated',
        updated_by: 'gaurav.singh@1mg.com',
      },
    },
  ],
  success: true,
};

export const mockFetchRavenActionsNode = {
  result: [
    {
      event: null,
      application: 'PHARMACY',
      required_fields: null,
      screen_text:
        'If you are not 100% satisfied with your purchase for any reason, you may initiate a return request for the delivered products. Make sure that the products are unused and in their original packaging.<br><br>Once your return request is received and verified your refund will be processed and automatically applied to your original method of payment within 7 days. <br><br>We encourage you to please check the package carefully before you accept delivery.<br><br>Know more about our <a href=https://tdgx9.app.goo.gl/?link=https://www.1mg.com/return-policy&al=onemg-web://www.1mg.com/return-policy&ibi=com.aranoah.healthkart&ifl=www.1mg.com/return-policy&apn=com.aranoah.healthkart.plus&afl=www.1mg.com/return-policy>Return Policy</a>',
      emails: null,
      optional_fields: null,
      type: 'INFO',
      id: 42,
      header: '1mg Return policy',
      response:
        '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.',
      updated_by: 'initialdbsetup@1mg.com',
      ticket: 1,
    },
  ],
  success: true,
};

export const mockFetchRavenTicket = {
  result: [
    {
      status: 'OPEN',
      priority: 'HIGH',
      updated_by: 'initialdbsetup@1mg.com',
      id: 1,
    },
    {
      status: 'OPEN',
      priority: 'LOW',
      updated_by: 'initialdbsetup@1mg.com',
      id: 2,
    },
    {
      status: 'OPEN',
      priority: 'MEDIUM',
      updated_by: 'initialdbsetup@1mg.com',
      id: 3,
    },
    {
      status: 'OPEN',
      priority: 'URGENT',
      updated_by: 'tushar.bansal@1mg.com',
      id: 4,
    },
  ],
  success: true,
};

export const mockFetchRavenMetaData = {
  result: {
    type: [
      {
        type: 'FORM',
        id: '1',
      },
      {
        type: 'INFO',
        id: '2',
      },
      {
        type: 'INFO_WITH_WRITE_TO_US',
        id: '3',
      },
      {
        type: 'WEBVIEW',
        id: '4',
      },
    ],
    fields: [
      {
        id: '1',
        fields: 'free_text',
      },
      {
        id: '2',
        fields: 'skus',
      },
      {
        id: '3',
        fields: 'attachments',
      },
    ],
  },
  success: true,
};

export const mockRavenNodes = {
  result: [
    {
      node_action: 177,
      updated_by: 'v-hritik.raj@1mg.com',
      application: 'PHARMACY',
      id: 473,
      name: 'eneter',
    },
    {
      node_action: 383,
      updated_by: 'deepanshu.kapoor@1mg.com',
      application: 'PHARMACY',
      id: 472,
      name: 'Tata NeuCoins le lo ji',
    },
  ],
  success: true,
};

export const mockCreateNodeEvent = {
  result: null,
  success: true,
};

export const mockGetLinkNodeEvents = {
  result: [
    {
      node_action: 88,
      name: 'I was overcharged for a product',
      application: 'PHARMACY',
      id: 89,
      updated_by: 'initialdbsetup@1mg.com',
    },
    {
      node_action: 88,
      name: 'I was overcharged for a product',
      application: 'PHARMACY',
      id: 68,
      updated_by: 'initialdbsetup@1mg.com',
    },
    {
      node_action: 88,
      name: 'Amount overcharged',
      application: 'PHARMACY',
      id: 48,
      updated_by: 'ankita.gupta@1mg.com',
    },
  ],
  success: true,
};
export const mockSendNodeLink = {
  result: null,
  success: true,
};
export const mockInactiveNodeLinks = {
  result: [
    '59.60',
    '1.161.462',
    '1.161.4.12',
    '1.161.4.13',
    '1.161.4.14',
    '1.161.4.17',
    '1.161.8.26',
    '1.161.8.27',
    '1.161.8.25',
    '1.161.6.22',
    '1.161.6.24',
    '1.161.6.23',
    '1.5.19',
    '151.107',
    '151.99',
    '151.101',
    '150.83',
    '151.96',
    '150.85',
    '150.85.91',
    '150.85.92',
    '150.85.93',
    '150.86.94',
    '150.86.95',
    '150.86',
    '150.87',
    '59.65',
    '150.82',
    '1.9.21',
    '58.38.55',
    '58.38.56',
    '58.38.57',
    '58.37.53',
    '58.37.54',
    '58.37',
    '58.34.41',
    '58.34.42',
    '58.34.43',
    '58.34.44',
    '58.34.45',
    '58.34.46',
    '58.34.47',
    '58.36.50',
    '58.36.51',
    '59.64.73',
    '59.64.74',
    '59.64.75',
    '59.62',
    '59.64',
    '148.76',
    '148.77.81',
    '148.77',
    '148.88',
    '1.161.462.460',
    '152.112.119',
    '154.143',
    '1.161.4.160',
    '1.161.2',
    '503.498.497',
    '1.161.6',
    '1.9.15',
    '1.9.18',
    '1.9.7',
    '1.163.164',
    '1.163.165',
    '1.5.20',
    '1.172',
    '152.112',
    '152.121',
    '152.112.120',
    '152.112.122',
    '152.113',
    '1.9.29',
    '152.116.125',
    '152.116.126',
    '152.116.127',
    '152.119',
    '152.111',
    '152.116',
    '151.101.109',
    '154.144',
    '151.98',
    '150.86.167',
    '150.90',
    '59.61.66',
    '59.61.67',
    '59.62.68',
    '59.62.69',
    '59.63.70',
    '59.63.71',
    '148.78',
    '148.166',
    '154.142',
    '155.145',
    '155.146',
    '229.220',
    '235.230',
    '246.236',
    '151.97.3',
    '253.247',
    '259.254',
    '1.229.220',
    '58.383',
    '263.260',
    '270.264',
    '271.270',
    '271.270.264',
    '280.272',
    '284.281',
    '288.285',
    '294.289',
    '1.307.306',
    '299.295',
    '307.306',
    '1.161.306',
    '311.309',
    '316.312',
    '1.470',
    '1.161.473',
    '1.471',
    '152.112.118',
    '1.278',
    '466.379',
    '151.97.103',
    '151.97.104',
    '151.101.108',
    '151.101.110',
    '321.317',
    '323.322',
    '1.161.455',
    '338.335',
    '151.97',
    '328.326',
    '333.330',
    '342.339',
    '347.343',
    '359.357',
    '352.349',
    '363.360',
    '367.364',
    '1.161.455.454',
    '150.378',
    '150.381',
    '1.383',
    '1.382',
    '58.381',
    '58.381.378',
    '1.380',
    '1.369',
    '1.378',
    '1.161.381.379',
    '1.161.4.379',
    '1.161.4.359',
    '1.161.4.356',
    '1.161.4.347.343',
    '1.474',
    '477.473',
    '479.453',
    '483.465',
    '153.134.383',
    '491.471',
    '1.491',
    '1.304',
    '1.283',
    '1.306',
    '1.42',
    '493.492',
    '493.492.491',
    '496.477',
    '499.495',
    '500.493',
    '150.381.378',
    '152.380',
    '58.378',
    '58.381.379',
    '58.370',
    '1.375',
    '58.375',
    '148.355',
    '1.347',
    '1.161.4.359.358',
    '1.161.4.358',
    '1.161.4.347',
    '474.472',
    '476.473',
    '478.473',
    '484.471',
    '485.468',
    '450.452.380',
    '1.491.471',
    '372.375',
    '500.493.492',
    '371.370',
    '381.378',
    '1.177',
    '150.381.379',
    '1.339',
    '1.377',
    '148.365',
    '152.382',
    '59.62.378',
    '1.161.4.359.356',
    '1.161.4.359.380',
    '1.161.4.359.354',
    '1.474.472',
    '475.473',
    '487.469',
    '450.380',
    '353.382',
    '1.374',
    '1.370',
    '372.377',
    '466.380',
    '495.490',
    '500.493.492.491',
    '502.496.477.473',
    '385.384',
    '388.387',
    '392.391',
    '393.392.391',
    '393.392.394',
    '393.392.395',
    '393.392.396',
    '401.399',
    '406.405',
    '408.407',
    '416.413',
    '420.417',
    '423.421',
    '426.424',
    '430.428',
    '433.431',
    '150.381.380',
    '58.380',
    '1.161.381',
    '1.294.377',
    '1.161.4.374',
    '1.161.4.359.357',
    '480.472',
    '481.471',
    '486.465',
    '153.134.177',
    '459.380',
    '1.161.491',
    '492.491.471',
    '372.378',
    '498.497',
    '499.495.490',
    '500.493.492.491.471',
    '503.498',
    '1.355',
    '1.161.4.355',
    '1.161.4.377',
    '1.161.4.359.377',
    '482.469',
    '152.316.380',
    '463.383',
    '1.161.491.471',
    '492.491',
    '493.492.491.471',
    '496.477.473',
    '502.496',
    '437.435',
    '440.438',
    '446.443',
    '148.77.80',
    '448.447',
    '452.451',
    '455.454',
    '457.456',
    '462.460',
    '463.462.460',
    '463.462.461',
    '463.464',
    '466.470',
    '497.494',
    '498.497.494',
    '502.496.477',
    '503.498.497.494',
    '59.441',
  ],
  success: true,
};
export const mockUpdateRankEvent = {
  result: null,
  success: true,
};
