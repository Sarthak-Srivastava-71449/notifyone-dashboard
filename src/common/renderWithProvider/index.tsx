export { default as withProvider } from './renderWithProvider';
export { store, withLocationProvider } from './renderWithProvider';
export {
  mockEmailEventDetails,
  mockPushNotificationEventDetails,
  mockSmsEventDetails,
  mockWhatsappEventDetails,
} from './constants';
