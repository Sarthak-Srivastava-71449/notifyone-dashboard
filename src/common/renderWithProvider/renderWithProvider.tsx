import React from 'react';
import thunk from 'redux-thunk';
import {
  applyMiddleware,
  combineReducers,
  legacy_createStore as createStore,
} from 'redux';
import { composeWithDevTools } from '@redux-devtools/extension';
import { Provider } from 'react-redux';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import { ToastContainer } from 'react-toastify';

import messages from '../i18n/messages';
import googleReducer from 'src/platform/reducers/auth.reducer';

import emailEventsReducer from 'src/store/reducers/emailEvents.reducer';
import pushNotificationEventsReducer from 'src/store/reducers/pushNotificationEvents.reducer';
import smsEventsReducer from 'src/store/reducers/smsEvents.reducer';
import whatsAppEventsReducer from 'src/store/reducers/whatsappEvents.reducer';

import Dashboard from 'src/view/components/Dashboard/Dashboard';
import CommunicationList from 'src/view/components/CommunicationList/CommunicationList';
import Communication from 'src/view/components/Communication/Communication';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

jest.mock('antd', () => {
  const antd = jest.requireActual('antd');

  return {
    ...antd,
  };
});

jest.mock('common/src/components/Spinner', () => ({
  Spinner: () => {
    return <></>;
  },
}));

let store: any;

const createTestStore = () => {
  const store = createStore(
    combineReducers({
      emailEvents: emailEventsReducer,
      pushNotificationEvents: pushNotificationEventsReducer,
      smsEvents: smsEventsReducer,
      whatsAppEvents: whatsAppEventsReducer,
      user: googleReducer,
    }),
    composeWithDevTools(applyMiddleware(thunk))
  );
  return store;
};

const withProvider = (component: any) => {
  store = createTestStore();
  store.dispatch = jest.fn();
  const locale = 'en-us';

  return (
    <Provider store={store}>
      <IntlProvider locale={locale} messages={messages[locale]}>
        <BrowserRouter>{component}</BrowserRouter>
      </IntlProvider>
    </Provider>
  );
};

const withLocationProvider = (to: string, state: any = {}) => {
  store = createTestStore();
  store.dispatch = jest.fn();
  const locale = 'en-us';

  window.history.pushState(state, '', to);

  return (
    <Provider store={store}>
      <IntlProvider locale={locale} messages={messages[locale]}>
        <BrowserRouter>
          <ToastContainer />
          <Routes>
            <Route path="communication" element={<Dashboard />}>
              <Route path="templates" element={<CommunicationList />} />
              <Route path="template" element={<Communication />} />
            </Route>
          </Routes>
        </BrowserRouter>
      </IntlProvider>
    </Provider>
  );
};

export { store, withLocationProvider };
export default withProvider;
