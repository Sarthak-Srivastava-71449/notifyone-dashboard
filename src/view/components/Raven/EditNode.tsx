import React, { Fragment, useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';
import { Input, Popconfirm } from 'antd';
import { Button } from '../Button';
import './Raven.css';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from 'src/store';
import { useNavigate, useParams } from 'react-router-dom';
import {
  fetchRavenRootNodes,
  updateNodeEvent,
} from 'src/store/actions/ravenRootNodeEvents';
import {
  getEditNodeDetails,
  getUpdateSuccess,
} from 'src/store/selectors/editAction';
import { getAccessToken } from 'src/store/selectors/accessToken';

const EditNode: React.FC = () => {
  /*
   Redux Selectors for fetching data
 */
  const nodeToEdit = useSelector((state: RootState) =>
    getEditNodeDetails(state)
  );
  const accessToken = useSelector((state: RootState) => getAccessToken(state));
  const updateSuccess = useSelector((state: RootState) =>
    getUpdateSuccess(state)
  );

  useEffect(() => {
    if (updateSuccess === true) {
      dispatch(fetchRavenRootNodes(accessToken, 200, 0));
      navigate('/communication/raven', { replace: true });
    }
  }, [updateSuccess]);

  const dispatch = useDispatch();

  const [nodeName, setNodeName] = useState<string>('');
  const { id } = useParams();
  const navigate = useNavigate();

  useEffect(() => {
    if (nodeToEdit === null || nodeToEdit?.id !== id) {
      navigate('/communication/raven', { replace: true });
    } else {
      setNodeName(nodeToEdit?.name);
    }
  }, [nodeToEdit]);

  /*
   Function for Submit update Data
  */
  const submitUpdateData = () => {
    const payload = {
      name: nodeName.trim(),
      node_action: null,
      node_id: String(nodeToEdit.id),
    };
    dispatch(updateNodeEvent(accessToken, payload));
  };

  const { TextArea } = Input;
  return (
    <Fragment>
      <div className="w-9/12 mt-12   mx-auto wallet-wrapper shadow-md">
        <h2 className="text-lg font-semibold raven-form-title text-center py-1">
          <FormattedMessage id="edit_node_form" />
        </h2>
        <div className="py-2 px-3 flex-col flex">
          <div className="flex mb-3 flex-col">
            <div className="input-label  flex w-full">
              <span
                className="text-sm node-name font-semibold whitespace-nowrap"
                data-testid="data-name"
              >
                <FormattedMessage id="node_name" />
                <span className="px-1">:</span>
              </span>
              <span className="w-full" data-testid="textArea">
                <TextArea
                  rows={4}
                  value={nodeName}
                  onChange={(e) => setNodeName(e.target.value)}
                />
              </span>
            </div>
            <div className="no-character flex">
              <div className="width-9"></div>
              <div className="width-91">
                {nodeName?.length > 0 && (
                  <span className="character">
                    <span className="text-sm  font-semibold">
                      <FormattedMessage id="no_of_characters" />
                    </span>{' '}
                    <span className="px-1">:</span>
                    {nodeName?.length}
                  </span>
                )}
              </div>
            </div>
          </div>
          <div className="flex justify-center items-center">
            <Button
              className="mr-1.5"
              onClick={() =>
                navigate('/communication/raven', { replace: true })
              }
            >
              <FormattedMessage id="back" />
            </Button>
            <Popconfirm
              className="popconfrim"
              title="Are you sureï¼Ÿ"
              onConfirm={() => submitUpdateData()}
              okText="Yes"
              cancelText="No"
            >
              <Button
                className=""
                disabled={
                  nodeName?.trim().length <= 1 ||
                  nodeName?.localeCompare(nodeToEdit?.name) === 0
                }
              >
                <FormattedMessage id="update" />
              </Button>{' '}
            </Popconfirm>
          </div>
        </div>
      </div>
    </Fragment>
  );
};
export default EditNode;
