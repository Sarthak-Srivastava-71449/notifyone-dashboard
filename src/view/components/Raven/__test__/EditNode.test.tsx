import React from 'react';
import { withProvider } from '../../../../common/renderWithProvider';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, screen } from '@testing-library/react';
import EditNode from '../EditNode';

jest.mock('src/store/actions/ravenNodeEvents');

describe('EditNode Test-Suite', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterEach(cleanup);
  test('Match Edit Node Snapshot', async () => {
    const { asFragment } = render(withProvider(<EditNode />));
    expect(asFragment()).toMatchSnapshot();
  });

  test('should span value matched to the document', () => {
    const component = render(withProvider(<EditNode />));
    component.getByTestId('data-name');
    expect(screen.queryByTestId('data-name')).toBeInTheDocument();
  });

  test('should text area value matched to the document', () => {
    const component = render(withProvider(<EditNode />));
    component.getByTestId('textArea');
    expect(screen.queryByTestId('textArea')).toBeInTheDocument();
  });
});
