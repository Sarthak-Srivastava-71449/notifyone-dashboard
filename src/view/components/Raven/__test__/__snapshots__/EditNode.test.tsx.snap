// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNode Test-Suite Match Edit Node Snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-9/12 mt-12   mx-auto wallet-wrapper shadow-md"
  >
    <h2
      class="text-lg font-semibold raven-form-title text-center py-1"
    >
      Edit Node Form
    </h2>
    <div
      class="py-2 px-3 flex-col flex"
    >
      <div
        class="flex mb-3 flex-col"
      >
        <div
          class="input-label  flex w-full"
        >
          <span
            class="text-sm node-name font-semibold whitespace-nowrap"
            data-testid="data-name"
          >
            Node Name
            <span
              class="px-1"
            >
              :
            </span>
          </span>
          <span
            class="w-full"
            data-testid="textArea"
          >
            <textarea
              class="ant-input"
              rows="4"
            />
          </span>
        </div>
        <div
          class="no-character flex"
        >
          <div
            class="width-9"
          />
          <div
            class="width-91"
          />
        </div>
      </div>
      <div
        class="flex justify-center items-center"
      >
        <button
          class="button primary-button mr-1.5"
          data-testid="custom-button"
        >
          Back
        </button>
        <span
          class="popconfrim"
        >
          <button
            class="button primary-button "
            data-testid="custom-button"
          >
            Update
          </button>
           
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
